yo1k/tic_tac_toe/test/core/test_game.py

```diff
@@ -22,242 +31,236 @@ def _new_state(
         self.assertEqual(state.step, 0)
         self.assertEqual(state.round, 1)
-        self.assertEqual(state.phase, g.Phase.INROUND)
+        self.assertEqual(state.phase, Phase.INROUND)
         self.assertEqual(state.players[0].wins, 1)
         self.assertEqual(state.players[1].wins, 0)
 
-    def test_advance_next_round_action_onset(self):
-        state = _new_state(phase=g.Phase.BEGINNING)
-        act_queue_px = ListActionQueue([g.Action.new_start()])
-        g.Logic([act_queue_px, []]).advance(state)
-        next_game_state_board = [[None, None, None], [None, None, None], [None, None, None]]
-        self.assertEqual(state.board.cells, next_game_state_board)
+    def test_advance__start_action__onset(self):
+        state = _new_state(phase=Phase.BEGINNING)
+        Logic([
+            ListActionQueue([Action.new_start()]),
+            ListActionQueue([None])]).advance(state)
+        expected_board_cells = [[None, None, None], [None, None, None], [None, None, None]]
+        self.assertEqual(state.board.cells, expected_board_cells)
         self.assertEqual(state.step, 0)
         self.assertEqual(state.round, 0)
-        self.assertEqual(state.phase, g.Phase.INROUND)
+        self.assertEqual(state.phase, Phase.INROUND)
         self.assertEqual(state.players[0].wins, 0)
         self.assertEqual(state.players[1].wins, 0)
 
 
 class LogicMultipleActionsTest(unittest.TestCase):
     def test_win(self):
-        state = _new_state(phase=g.Phase.BEGINNING)
+        state = _new_state(phase=Phase.BEGINNING)
         act_queue_px = ListActionQueue([
-            g.Action.new_start(),
-            g.Action.new_occupy(g.Cell(1, 1)),
+            Action.new_start(),
+            Action.new_occupy(Cell(1, 1)),
             None,
-            g.Action.new_occupy(g.Cell(0, 0)),
-            g.Action.new_occupy(g.Cell(0, 2)),
-            g.Action.new_occupy(g.Cell(2, 0))])
+            Action.new_occupy(Cell(0, 0)),
+            Action.new_occupy(Cell(0, 2)),
+            Action.new_occupy(Cell(2, 0))])
         act_queue_po = ListActionQueue([
-            g.Action.new_occupy(g.Cell(1, 2)),
-            g.Action.new_occupy(g.Cell(2, 2)),
+            Action.new_occupy(Cell(1, 2)),
+            Action.new_occupy(Cell(2, 2)),
             None,
-            g.Action.new_occupy(g.Cell(0, 1))])
+            Action.new_occupy(Cell(0, 1))])
         expected_board_cells = [
-            [g.Mark.X, g.Mark.O, g.Mark.X],
-            [None, g.Mark.X, g.Mark.O],
-            [g.Mark.X, None, g.Mark.O]]
-        i = 0
-        while i <= len(act_queue_px.actions) + len(act_queue_po.actions):
-            g.Logic([act_queue_px, act_queue_po]).advance(state)
-            i += 1
+            [Mark.X, Mark.O, Mark.X],
+            [None, Mark.X, Mark.O],
+            [Mark.X, None, Mark.O]]
+        for i in range(0, len(act_queue_px.actions) + len(act_queue_po.actions)):
+            Logic([act_queue_px, act_queue_po]).advance(state)
```

Comment:
A new instance of `Logic` should not be created on each loop iteration. The same is applicable to `test_draw`.

Reply:
Done.